block/config.ml

```diff
@@ -2,9 +2,10 @@ open Mirage
 
 let main = foreign "Unikernel.Main" (console @-> block @-> job)
 
-let img = match get_mode () with
-  | `Xen -> block_of_file "xvda1"
-  | `Unix | `MacOSX -> block_of_file "disk.img"
+let img =
+  if_impl Key.is_xen
+    (block_of_file "xvda1")
+    (block_of_file "disk.img")
```

Comment:
Why is `get_mode` bad? As we add more platforms (`Javascript`, `Rump`, etc) using `if` is likely to break.

I was wondering about having the user provide the list of supported platforms in the `config.ml`, e.g.

    let mode = get_mode [`Xen; `Unix]

This would set `mode` to one of the modes in the list, or report an error if the user picked a mode that wasn't supported (that avoids users having to write default match cases everywhere for modes they don't want to support).